src/couch_replicator_doc_processor.erl

```diff
@@ -104,14 +112,15 @@ process_change(DbName, {Change}) ->
         undefined ->
             ok = process_updated(Id, JsonRepDoc);
         <<"triggered">> ->
-            couch_replicator_docs:remove_state_fields(DbName, DocId),
+            maybe_remove_state_fields(DbName, DocId, compat_mode()),
```

Comment:
I'd prefer `maybe_remove_state_fields(DbName, DocId)` here and then a `maybe_remove_state_fields/2` that calls compat_mode() then calls `maybe_remove_state_fields/3`.